include/silo/common/fasta_reader.h

```diff
@@ -0,0 +1,24 @@
+#ifndef SILO_FASTA_READER_H
+#define SILO_FASTA_READER_H
+
+#include "iostream"
+
+#include "silo/common/input_stream_wrapper.h"
+
+namespace silo {
+class FastaReader {
+  private:
+   silo::InputStreamWrapper in_file;
+
+  public:
+   explicit FastaReader(const std::string& in_file_name);
```

Comment:
Can we use the filesystem::path here instead of string? This would make a test, if the path exists a little easier. Maybe also add the exception in the cpp if it does not exist, or can we cope with a not provided file?

Reply:
Maybe we could also add this into the InputStreamWrapper, or is it nested to deeply?